src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -772,7 +774,90 @@ public String getFileType(final Map<String, String> metaData, final String forms
     }
 
     /**
-     * Parameters that are used to determine filetype in {@link #getFileType(Map, String)}
+     * Get the file type from the IBaseDataObject passed in
+     *
+     * @param bdo IBaseDataObject
+     * @return the file type (also added to the IBaseDataObject if not present)
+     */
+    public static String getFileType(final IBaseDataObject bdo) {
+        return getFileType(bdo, null);
+    }
+
+    /**
+     * Get the file type from the IBaseDataObject or the form string passed in
+     *
+     * @param bdo IBaseDataObject
+     * @param formsArg space separated string of current forms
+     * @return the file type
+     */
+    public static String getFileType(final IBaseDataObject bdo, final String formsArg) {
+        String forms = formsArg;
+        if (forms == null) {
+            forms = bdo.getStringParameter(FileTypeCheckParameter.POPPED_FORMS.getFieldName());
+            if (forms == null) {
+                forms = "";
+            }
+        }
+
+        String fileType;
+        if (bdo.hasParameter(FileTypeCheckParameter.FILETYPE.getFieldName())) {
+            fileType = bdo.getStringParameter(FileTypeCheckParameter.FILETYPE.getFieldName());
+        } else if (bdo.hasParameter(FileTypeCheckParameter.FINAL_ID.getFieldName())) {
+            fileType = bdo.getStringParameter(FileTypeCheckParameter.FINAL_ID.getFieldName());
+            logger.debug("FINAL_ID FileType is ({})", fileType);
+        } else {
+            if (forms.contains(" ")) {
+                fileType = forms.substring(0, forms.indexOf(" ")).trim();
+            } else {
+                fileType = forms;
+            }
+            if (StringUtils.isEmpty(fileType)) {
+                if (bdo.hasParameter(FileTypeCheckParameter.FONT_ENCODING.getFieldName())) {
+                    fileType = "TEXT";
+                } else {
+                    fileType = "UNKNOWN";
+                }
+            }
+        }
+
+        if ("UNKNOWN".equals(fileType) && forms.contains("MSWORD")) {
+            fileType = "MSWORD_FRAGMENT";
+        }
+
+        if ("QUOTED-PRINTABLE".equals(fileType) || fileType.startsWith("LANG-") || fileType.startsWith("ENCODING(")) {
+            fileType = "TEXT";
+        }
+
+        return fileType;
+    }
+
+    /**
+     * Get the complete file type from the IBaseDataObject or the form string passed in
+     *
+     * @param bdo IBaseDataObject
+     * @param formsArg space separated string of current forms
+     * @return the complete file type
+     */
+    public static String getCompleteFileType(final IBaseDataObject bdo, final String formsArg) {
```

Comment:
I'm a bit puzzled by this method name.  Why `getCompleteFileType` as the method name, but completely ignore any bdo parameter stored under the key `FileTypeCheckParameter.COMPLETE_FILETYPE.getFieldName()`?  Admittedly, these are a bit hard to grok since the documentation re: expected values stored under of various parameter keys is very sparse in this codebase.